frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_user_view.dart

```diff
@@ -49,28 +50,84 @@ class SettingsUserView extends StatelessWidget {
             crossAxisAlignment: CrossAxisAlignment.start,
             mainAxisSize: MainAxisSize.min,
             children: [
-              _renderUserNameInput(context),
-
+              _buildUserIconSetting(context),
               if (isSupabaseEnabled) ...[
-                const VSpace(20),
+                const VSpace(12),
                 UserEmailInput(user.email)
               ],
-
-              const VSpace(20),
-              _renderCurrentIcon(context),
-              const VSpace(20),
+              const VSpace(12),
               _renderCurrentOpenaiKey(context),
-              const VSpace(20),
-              // _renderHistoricalUser(context),
+              const VSpace(12),
               _renderLoginOrLogoutButton(context, state),
-              const VSpace(20),
+              const VSpace(12),
             ],
           ),
         ),
       ),
     );
   }
 
+  Row _buildUserIconSetting(BuildContext context) {
+    return Row(
+      children: [
+        UserAvatar(
+          iconUrl: user.iconUrl,
+          name: user.name,
+          isLarge: true,
+        ),
+        const HSpace(8),
+        Column(
+          children: [
+            FlowyIconButton(
+              width: 24,
+              tooltipText: LocaleKeys.settings_user_tooltipRemoveIcon.tr(),
+              icon: const FlowySvg(FlowySvgs.restore_s),
+              onPressed: () => context
+                  .read<SettingsUserViewBloc>()
+                  .add(const SettingsUserEvent.updateUserIcon()),
+            ),
+            const VSpace(4),
+            FlowyIconButton(
+              width: 24,
+              tooltipText: LocaleKeys.settings_user_tooltipSelectIcon.tr(),
+              icon: const FlowySvg(FlowySvgs.emoji_s),
+              onPressed: () => showDialog(
+                context: context,
+                builder: (dialogContext) {
+                  return SimpleDialog(
+                    title: FlowyText.medium(
+                      LocaleKeys.settings_user_selectAnIcon.tr(),
+                      fontSize: FontSizes.s16,
+                    ),
+                    children: [
+                      SizedBox(
+                        height: 300,
+                        width: 300,
+                        child: IconGallery(
+                          user.iconUrl,
+                          (iconUrl, isSelected) {
+                            context.read<SettingsUserViewBloc>().add(
+                                  SettingsUserEvent.updateUserIcon(
+                                    iconUrl: isSelected ? "" : iconUrl,
```

Comment:
I think clicking on the currently-selected user icon to de-select it is strange when there is a reset button.